wfe2/wfe.go

```diff
@@ -2594,62 +2632,18 @@ func (wfe *WebFrontEndImpl) RenewalInfo(ctx context.Context, logEvent *web.Reque
 	serial := certID.Serial()
 	logEvent.Extra["RequestedSerial"] = serial
 
-	sendRI := func(ri core.RenewalInfo) {
-		response.Header().Set(headerRetryAfter, fmt.Sprintf("%d", int(6*time.Hour/time.Second)))
-		err := wfe.writeJsonResponse(response, logEvent, http.StatusOK, ri)
-		if err != nil {
-			wfe.sendError(response, logEvent, probs.ServerInternal("Error marshalling renewalInfo"), err)
-			return
-		}
-	}
-
-	// Check if the serial is part of an ongoing/active incident, in which case
-	// the client should replace it now.
-	result, err := wfe.sa.IncidentsForSerial(ctx, &sapb.Serial{Serial: serial})
-	if err != nil {
-		wfe.sendError(response, logEvent, web.ProblemDetailsForError(err,
-			"checking if certificate is impacted by an incident"), err)
+	renewalInfo, prob, err := wfe.determineARIWindow(ctx, serial)
```

Comment:
I don't love the pattern of returning a ProblemDetails _and_ an error. Looking at the factored-out `determineARIWindow`, I see that all error returns are of the general category "internal server error", _except_ for one NotFound case.

It would be better to have `determineARIWindow` return `(core.RenewalInfo, error)`, pass through the error return from `sa.GetCertificate`, and do the `ProblemDetailsForError` dance here in `NewOrder`, on line 2637. I think that's more consistent with what we do elsewhere in the WFE, and it means we don't have to do `ProblemDetailsForError` in each return site within `determineARIWindow`.

Reply:
I've gone ahead with a refactor which does the problem details once, but inside of this method. I don't like the idea of having to handle the the `ProblemDetailsForError` _**or**_ `probs.NotFound` dance here inside of `wfe.RenewalInfo()` and elsewhere inside of `wfe.NewOrder()` in #7298.

Reply:
My point with this is twofold:

 - We only need one return value to represent an error. Returning both `ProblemDetails` _and_ an error invites logic bugs and doesn't buy us anything.
 - For that one return value, it's better to return an `error` than a `ProblemDetails` for a couple of reasons.
   - Avoid boxed `nil` bugs
   - This allows `determineARIWindow` to be a "boring" function, where errors are returned directly to the layer above, with no annotation as to specific problem types. Which is nice because `determineARIWindow` doesn't really have any specific problem types it wants to add.